src/libraries/System.Security.Cryptography.Encoding/src/System/Security/Cryptography/PemEncoding.cs

```diff
@@ -0,0 +1,518 @@
+                                   out int base64end,
+                                   out int decodedSize))
+                {
+                    goto NextAfterLabel;
+                }
+
+                Range pemRange = preebIndex..pemEndIndex;
+                Range base64range = (contentStartIndex + base64start)..
+                                    (contentStartIndex + base64end);
+                fields = new PemFields(labelRange, base64range, pemRange, decodedSize);
+                return true;
+
+                NextAfterLabel:
+                if (preebEndIndex <= 0)
+                {
+                    // We somehow ended up in a situation where we will advance
+                    // 0 or -1 characters, which means we'll probably end up here again,
+                    // advancing 0 or -1 characters, in a loop. To avoid getting stuck,
+                    // detect this situation and return.
+                    fields = default;
+                    return false;
+                }
+                areaOffset += preebEndIndex;
+            }
+
+            fields = default;
+            return false;
+
+            static ReadOnlySpan<char> WritePostEB(ReadOnlySpan<char> label, Span<char> destination)
+            {
+                int size = PostEBPrefix.Length + label.Length + Ending.Length;
+                Debug.Assert(destination.Length >= size);
+                PostEBPrefix.AsSpan().CopyTo(destination);
+                label.CopyTo(destination[PostEBPrefix.Length..]);
+                Ending.AsSpan().CopyTo(destination[(PostEBPrefix.Length + label.Length)..]);
+                return destination[..size];
+            }
+        }
+
+        private static int IndexOfByOffset(this ReadOnlySpan<char> str, ReadOnlySpan<char> value, int startPosition)
+        {
+            int index = str[startPosition..].IndexOf(value);
+            return index == -1 ? -1 : index + startPosition;
+        }
+
+        private static bool IsValidLabel(ReadOnlySpan<char> data)
+        {
+            static bool IsLabelChar(char c) => (uint)(c - 0x21u) <= 0x5du && c != '-';
+
+            // Empty labels are permitted per RFC 7468.
+            if (data.IsEmpty)
+                return true;
+
+            // First character of label must be a labelchar, which is a character
+            // in 0x21..0x7e (both inclusive), except hyphens.
+            if (!IsLabelChar(data[0]))
+                return false;
+
+            bool previousSpaceOrHyphen = false;
+            for (int index = 1; index < data.Length; index++)
+            {
+                char c = data[index];
+
+                if (IsLabelChar(c))
+                {
+                    previousSpaceOrHyphen = false;
+                    continue;
+                }
+
+                bool isSpaceOrHyphen = c == ' ' || c == '-';
+
+                if (!isSpaceOrHyphen || previousSpaceOrHyphen)
+                    return false;
```

Comment:
Unbraced statements aren't allowed within braced statements.  (Which has a net result of: they're only allowed in the outer method scope)

```suggestion
                {
                    return false;
                }
```

Reply:
@bartonjs okay, I made this mistake a number of times. I will address all of them.